### ISSUE TITLE
patchRequire doesn't work on Windows?

### ISSUE THREAD

**Issue by fearthecowboy** [streamich/fs-monkey#1]
First: AWESOME -- I was thinking about writing this exact same thing, but I'm happy that you did such a great job on this and it's friends (memfs, unionfs, etc...)

Now, down to brass tacks -- 

it would appear that `patchRequire` doesn't work on Windows -my guess being that that paths via memfs are always translated into unix-style and the code in `require`  knows that on windows it has to do windows-specific work.

I was going to debug into this, but any knowledge you had would be wonderful

**Comment by streamich** [comment]
I have published #2 as version `0.2.1`, but made the path rewrite op-in with `unixifyPaths`:

https://github.com/streamich/fs-monkey/blob/master/src/patchRequire.js#L36

**Comment by fearthecowboy** [comment]
Sweet! 👍

**Comment by fearthecowboy** [comment]
@streamich Dude. Way smart move with making that opt-in. turns out that I have use cases for both ways.

You deserve a free beer:

🍺 

😆

**Comment by streamich** [comment]
What are your uses cases?

**Comment by fearthecowboy** [comment]
Well, my primary use case was to effectively 'static-link' stuff into a node app, and stuff all the modules down to a single file.  I used this is in the [dotnet-sdk-2.0.0](https://www.npmjs.com/package/dotnet-sdk-2.0.0) and [dotnet-2.0.0](https://www.npmjs.com/package/dotnet-2.0.0) packages to embed `yarn` (and it's dependencies) instead of having it as a regular node dependency.  This made it possible to drop the install time of dotnet-runtime and dotnet-sdk down to between 3-7 seconds on any platform (my previous package could take upwards of 50 seconds on Windows, because Windows Defender has a significant per-file overhead when scanning during disk access) Using a single file to store the node_modules content still allows defender to scan it, but it's nearly instantaneous.

In that case, I never wanted the possibility that files could be requested from disk.

After getting such a boost for that, I turned to doing the same for some other projects, except they have more than a single file on disk for the node module, so I still need to fall back to `node_modules` instead of having it always pulling from the static file system.  In that case, I actually need to preserve the original path so that I can pull thru to the real file system.

In my testing of that scenario, I uncovered one glitch, which I'm tracking down right now--if you run the app from a different drive letter than where the app is installed on disk, relative module resolution is losing the drive letter, and looking on the local drive. 

Once I'm finished hashing this out, I think I can distill it all down to a little tool that can be called after build time to 'static-link' a given set of node dependencies.
